plugins/rebalance/rest.go

```diff
@@ -0,0 +1,279 @@
+package rebalance
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// RebalanceStatus represents Rebalance Status
+type RebalanceStatus uint64
+
+const (
+	// Started should be set only for a volume that has been just started rebalance process
+	Started RebalanceStatus = iota
+	// Inprogress should be set only for a volume that are running rebalance process
+	Inprogress
+	// Failed should be set only for a volume that are failed to run rebalance process
+	Failed
+	// Completed should be set only for a volume that the rebalance process is completed
+	Completed
+	// Stopped should be set only for a volume that has been just stopped rebalance process
+	Stopped
+)
+
+// RebalanceInfo represents Rebalance details
+type RebalanceInfo struct {
+	Volname           string
+	Status            RebalanceStatus
+	RebalanceID       uuid.UUID
+	RebalanceFiles    uint64
+	RebalanceData     uint64
+	LookedupFiles     uint64
+	RebalanceFailures uint64
+	RebalanceTime     uint64
+	SkippedFiles      uint64
+	CommitHash        uint64
```

Comment:
I don't think the CommitHash is saved in this. However, looks like some of the time fields (time_left) are missing.
Please rename RebalanceTime to ElapsedTime.